frame/support/src/storage/bounded_vec.rs

```diff
@@ -296,6 +296,13 @@ impl<T, S> Deref for BoundedVec<T, S> {
 	}
 }
 
+// Allows for mutable borrow operations of `Vec<T>` on `BoundedVec<T>`.
+impl<T, S> DerefMut for BoundedVec<T, S> {
```

Comment:
woooooaaaaa no way I can allow this :D this allows you to do all operations that require `&mut self` on the bounded vector, like `push`, `pop` etc. 

If you want something like `last_mut`, you can add it manually, if you can argue in the comments that it can never lead to the size of the bounded vec to be increased.

Reply:
maybe we need a trait of BoundedDerefMut ops - so ops that need a mutable deref, but will never grow a collection

Reply:
for instance `pop` or `last_mut` won't grow a collection

Reply:
lmk if the changes in c876504 & a2638f1 look ok

Reply:
all good now.